react/AppLinker/index.spec.jsx

```diff
@@ -73,20 +75,25 @@ describe('app icon', () => {
   })
 
   it('should render correctly', () => {
-    const root = shallow(<AppItem t={tMock} app={app} />).dive()
-    expect(root.getElement()).toMatchSnapshot()
+    const { container } = setup({ app })
+    expect(container).toMatchSnapshot()
   })
 
-  it('should work for native -> native', () => {
-    const root = shallow(
-      <AppItem t={tMock} app={app} onAppSwitch={appSwitchMock} />
-    ).dive()
-    root.find('a').simulate('click')
+  it('should work for web -> web', async () => {
+    isMobileApp.mockReturnValue(false)
+    const { container, user } = setup({ app, onAppSwitch: appSwitchMock })
+    const link = container.querySelector('a')
+    await user.click(link)
```

Comment:
Why do we need to await the click? What are the async stuff we're doing in that case?

Reply:
In this specific example (`should work for web -> web`) there was no need to await for click result.

The original need was for `should work for native -> native` test. With this configuration, AppLinker calls `AppLinker.openNativeFromNative()` which then calls `startApp()` which is a promise.

I'm not sure why, but in that case, when calling `fireEvent.click(link)` then the check `expect(startApp).toHaveBeenCalledWith` failed.

My intuition was that it wouldn't fail because `startApp` should be called before the method returns (event if it does not wait its result). But in practice: it appears that it is not the case. Maybe this is a race condition?

However, `user.click()` bring the `act()` mechanism by default which seems to fix our problem. The implication is that `user.click()` is a promise, and so we have to await it.

For all other tests, I used `user.click` for homogenisation purpose.